source/javascripts/components/unified-editor/WorkflowConfig/WorkflowConfigPanel.tsx

```diff
@@ -1,19 +1,36 @@
 import { TabPanel, TabPanels, Tabs } from '@bitrise/bitkit';
+import { useFormContext } from 'react-hook-form';
+import { useShallow } from 'zustand/react/shallow';
+import useSearchParams from '@/hooks/useSearchParams';
+import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
 import WorkflowConfigHeader from './components/WorkflowConfigHeader';
 import ConfigurationTab from './tabs/ConfigurationTab';
 import PropertiesTab from './tabs/PropertiesTab';
 import WorkflowConfigProvider from './WorkflowConfig.context';
-import { WorkflowConfigTab } from './WorkflowConfig.types';
-
-type Props = {
-  workflowId: string;
-};
+import { FormValues, WorkflowConfigTab } from './WorkflowConfig.types';
+import useRenameWorkflow from './hooks/useRenameWorkflow';
 
 const WorkflowConfigPanelContent = () => {
+  const form = useFormContext<FormValues>();
+  const originalName = form.formState.defaultValues?.properties?.name ?? '';
+  const [, setSearchParams] = useSearchParams();
+  const updateWorkflow = useBitriseYmlStore(useShallow((s) => s.updateWorkflow));
+  const renameWorkflow = useRenameWorkflow((newWorkflowId) => {
+    setSearchParams((searchParams) => ({
```

Comment:
Do we need this complicated hook to debounce the setSearchParam?

Reply:
Without debounce, the name input has a serious input lag because renaming workflow is a heavy process with lot of side-effects... So the debounce relieves the serious input lag.